src/services/mqtt/mqtt-broker.client.test.ts

```diff
@@ -0,0 +1,93 @@
+import mqtt from 'mqtt';
+import * as sinon from 'sinon';
+import { suite } from 'uvu';
+import * as assert from 'uvu/assert';
+
+import { MqttBrokerClient, MqttSubscribePayload } from './mqtt-broker.client';
+
+const test = suite('MQTT Broker Client');
+
+test.before.each(async (context) => {
+  context.sandbox = sinon.createSandbox();
+});
+
+test.after.each((context) => {
+  context.sandbox.restore();
+});
+
+test('should call connect', async (context) => {
+  const onSpy = context.sandbox.spy((event: string, cb: () => void) => cb());
+  const mqttConnectStub = context.sandbox.stub(mqtt, 'connect').returns({
+    on: onSpy,
+  });
+
+  const client = new MqttBrokerClient('mqtt://localhost:1883');
+  await client.connect();
+
+  assert.is(mqttConnectStub.called, true);
+  assert.is(onSpy.called, true);
+});
+
+test('should call subscribe', async (context) => {
+  const subscribeFunctionSpy = context.sandbox.spy((event: MqttSubscribePayload) => event);
+  const onSpy = context.sandbox.spy((event: string, cb: () => void) => cb());
```

Comment:
I am not sure it makes sense spying on a function we create on the fly. Think this could use `.fake()` throughout the whole test file.
```suggestion
  const onSpy = context.sandbox.fake();
```